src/modules/lua.c

```diff
@@ -73,11 +73,17 @@ mtev_lua_lmc_alloc(mtev_dso_generic_t *self, mtev_lua_resume_t resume) {
 void
 mtev_lua_lmc_free(lua_module_closure_t *lmc) {
   if(lmc) {
+    if(lmc->lua_state) lua_close(lmc->lua_state);
```

Comment:
In that case, why do you need the lock at all? AFAICT, it exists solely to protect `mtev_lua_states` from access from multiple threads, but if everything's in one thread, that'd be unnecessary? Or I may have miscommunicated... I'm not worried about multiple threads talking to `lmc`, but about multiple threads talking to `lmc->lua_state`, which is accessible from other threads via that hash-table.